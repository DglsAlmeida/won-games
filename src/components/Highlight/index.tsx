import Button from 'components/Button'
import * as S from './styles'

export type HighlightProps = {
  title: string
  subtitle: string
  buttonLink: string
  backgroundImage: string
  buttonLabel: string
  floatImage?: string
  alignment?: 'right' | 'left'
}

const Highlight = ({
  title,
  subtitle,
  buttonLabel,
  backgroundImage,
  floatImage,
  alignment = 'right',
  buttonLink
}: HighlightProps) => (
  <S.Wrapper backgroundImage={backgroundImage} alignment={alignment}>
    {!!floatImage && <S.FloatImage src={floatImage} alt={title} />}
    <S.Content>
      <S.Title>{title}</S.Title>
      <S.Subtitle>{subtitle}</S.Subtitle>
      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </S.Content>
  </S.Wrapper>
)

export default Highlight
