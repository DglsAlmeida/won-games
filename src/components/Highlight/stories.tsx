import { Meta, Story } from '@storybook/react'
import Highlight, { HighlightProps } from '.'

export default {
  title: 'Highlight',
  component: Highlight
} as Meta

export const Basic: Story<HighlightProps> = (args) => (
  <div style={{ maxWidth: '104rem' }}>
    <Highlight {...args} />
  </div>
)

Basic.args = {
  title: "Read Dead it's back",
  subtitle: "Come see John's new adventures",
  buttonLink: '/test',
  backgroundImage: '/img/red-dead-img.jpeg',
  buttonLabel: 'Buy now'
}

export const WithImage: Story<HighlightProps> = (args) => (
  <div style={{ maxWidth: '104rem' }}>
    <Highlight {...args} />
  </div>
)

WithImage.args = {
  title: "Read Dead it's back",
  subtitle: "Come see John's new adventures",
  buttonLink: '/test',
  backgroundImage: '/img/red-dead-img.jpeg',
  buttonLabel: 'Buy now',
  floatImage: 'img/red-dead-float.png'
}
