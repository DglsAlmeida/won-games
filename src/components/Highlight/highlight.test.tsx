import { screen } from '@testing-library/react'
import Highlight, { HighlightProps } from './index'
import { renderWithTheme } from '../../utils/tests/helpers'
import * as S from './styles'

const props: HighlightProps = {
  title: 'Heading 1',
  subtitle: 'Heading 2',
  buttonLink: '/test',
  backgroundImage: '/img/red-dead-img.jpeg',
  floatImage: '/float-image.png',
  buttonLabel: 'Buy now'
}

describe('<Highlight />', () => {
  it('should render the headings and button', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(
      screen.getByRole('heading', { name: /heading 1/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /heading 2/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /buy now/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render background img', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(container.firstChild).toBeInTheDocument()

    expect(container.firstChild).toHaveStyle({
      backgroundImage: `url(${props.backgroundImage})`
    })
  })

  it('should render float img', () => {
    renderWithTheme(<Highlight {...props} floatImage={props.floatImage} />)

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      '/float-image.png'
    )
  })

  it('should render align right by default', () => {
    const { container } = renderWithTheme(
      <Highlight {...props} floatImage={props.floatImage} />
    )

    expect(container.firstChild).toHaveStyleRule(
      'grid-template-areas',
      "'floatImage content'"
    )

    expect(container.firstChild).toHaveStyleRule('text-align', 'right', {
      modifier: `${S.Content}`
    })
  })

  it('should render align left', () => {
    const { container } = renderWithTheme(
      <Highlight {...props} floatImage={props.floatImage} alignment="left" />
    )

    expect(container.firstChild).toHaveStyleRule(
      'grid-template-areas',
      "'content floatImage'"
    )
  })
})
