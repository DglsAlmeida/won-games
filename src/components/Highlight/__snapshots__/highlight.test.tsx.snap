// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render the headings and button 1`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c6:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  position: relative;
  height: 23rem;
  display: grid;
  background-image: url(/img/red-dead-img.jpeg);
  background-position: center center;
  background-size: cover;
  grid-template-areas: 'floatImage content';
  grid-template-columns: 1.3fr 2fr;
}

.c0:after {
  content: '';
  position: absolute;
  height: 100%;
  width: 100%;
  background-color: rgba(0,0,0,0.6);
}

.c0 .c2 {
  text-align: right;
}

.c1 {
  grid-area: floatImage;
  z-index: 10;
  max-height: 23rem;
  max-width: 100%;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
}

.c3 {
  grid-area: content;
  z-index: 10;
  padding: 1.6rem;
}

.c4 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c5 {
  font-size: 1.4rem;
  font-weight: 300;
  color: #FAFAFA;
  margin-bottom: 3.2rem;
}

@media (min-width:768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c1 {
    max-height: 32rem;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 1.8rem;
  }
}

<section
  class="c0"
>
  <img
    alt="Heading 1"
    class="c1"
    src="/float-image.png"
  />
  <div
    class="c2 c3"
  >
    <h2
      class="c4"
    >
      Heading 1
    </h2>
    <h3
      class="c5"
    >
      Heading 2
    </h3>
    <a
      class="c6"
      href="/test"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</section>
`
